Mathlib/Algebra/Module/Submodule/IsometryMap.lean

```diff
@@ -0,0 +1,53 @@
+/-
+Copyright (c) 2025 Jason Reed. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jason Reed
+-/
+import Mathlib.Analysis.Normed.Operator.LinearIsometry
+
+/-!
+# Restriction of linear isometries (and equivalences) to `Submodule`s.
+
+## Main declarations
+
+* `LinearIsometry.submoduleMap`: A linear isometry restricted to a submodule
+* `LinearIsometryEquiv.submoduleMap`: A linear isometry equivalence restricted to a submodule
+
+## Tags
+
+submodule, subspace, linear map, isometry, equivalence
+-/
+
+variable {R : Type*} {R₁ : Type*} {R₂ : Type*}
+variable {M : Type*} {M₁ : Type*} {M₂ : Type*}
+
+namespace LinearIsometry
+
+variable [Ring R] [SeminormedAddCommGroup M] [SeminormedAddCommGroup M₁]
+variable [Module R M] [Module R M₁]
+
+/-- A linear isometry between two modules restricts to a linear isometry from any submodule p of the
+domain onto the image of that submodule.
+
+This is a version of LinearMap.submoduleMap extended to linear isometries -/
+def submoduleMap (p : Submodule R M) (e : M →ₗᵢ[R] M₁) :
```

Comment:
```suggestion
@[simps]
def submoduleMap (p : Submodule R M) (e : M →ₗᵢ[R] M₁) :
```
should generate a helper lemma, which you can check the statement of by hovering over `simps`.